transport/http/handler.go

```diff
@@ -67,9 +68,13 @@ func (h handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	if status.Name() != "" {
 		responseWriter.AddSystemHeader(ErrorNameHeader, status.Name())
 	}
-	// TODO: would prefer to have error message be on a header so we can
-	// have non-nil responses with errors, discuss
-	_, _ = fmt.Fprintln(responseWriter, status.Message())
+	if bothResponseError {
+		responseWriter.AddSystemHeader(BothResponseErrorHeader, AcceptTrue)
+		responseWriter.AddSystemHeader(ErrorMessageHeader, status.Message())
+	} else {
+		responseWriter.ResetBuffer()
```

Comment:
won't the buffer get "reset" once we close the responseWriter?  Isn't this redundant?

Reply:
This just clears out any data that has been written to the responseWriter already, otherwise if we have written data to the responseWriter, we will write `[body][error]` to the response